commit 1ae0268dc54e6657605216669eefab4268165237
Author: Stefan Agner <stefan@agner.ch>
Date:   Sun Jun 17 12:33:50 2018 +0200

    brcmsmac: fix wrap around in conversion from constant to s16
    
    [ Upstream commit c9a61469fc97672a08b2f798830a55ea6e03dc4a ]
    
    The last value in the log_table wraps around to a negative value
    since s16 has a value range of -32768 to 32767. This is not what
    the table intends to represent. Use the closest positive value
    32767.
    
    This fixes a warning seen with clang:
    drivers/net/wireless/broadcom/brcm80211/brcmsmac/phy/phy_qmath.c:216:2: warning:
          implicit conversion from 'int' to 's16' (aka 'short') changes
    value from 32768
          to -32768 [-Wconstant-conversion]
            32768
            ^~~~~
    1 warning generated.
    
    Fixes: 4c0bfeaae9f9 ("brcmsmac: fix array out-of-bounds access in qm_log10")
    Cc: Tobias Regnery <tobias.regnery@gmail.com>
    Signed-off-by: Stefan Agner <stefan@agner.ch>
    Signed-off-by: Kalle Valo <kvalo@codeaurora.org>
    Signed-off-by: Sasha Levin <alexander.levin@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmsmac/phy/phy_qmath.c b/drivers/net/wireless/broadcom/brcm80211/brcmsmac/phy/phy_qmath.c
index b9672da24a9d..b24bc57ca91b 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmsmac/phy/phy_qmath.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmsmac/phy/phy_qmath.c
@@ -213,7 +213,7 @@ static const s16 log_table[] = {
 	30498,
 	31267,
 	32024,
-	32768
+	32767
 };
 
 #define LOG_TABLE_SIZE 32       /* log_table size */
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1178_1ae0268dc54e6657605216669eefab4268165237.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1178_1ae0268dc54e6657605216669eefab4268165237.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
